test/functional/kernel/basic-single-loop/tests/test-kernel-resource-basic-single-loop-segments.hpp

```diff
@@ -0,0 +1,138 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
```

Comment:
How much overlap is there between this file and the non-resource test file? Can that be reduced and invoke the different tests using a simple template parameter type?

Reply:
They're fairly similar, I'll have a go at consolidating the two test implementations into one.

Reply:
If it starts to get too convoluted, stick with what you have.

Reply:
Sounds good, I'll let you know